### PR TITLE
BLD: Move options to pyproject.toml

### PR DESCRIPTION
**Pull request by lithomas1**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

Spat out from #49115.
Versioneer needed an upgrade to read from pyproject.toml, but the update should be pretty low-risk/benign.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -5,12 +5,145 @@ requires = [
     "setuptools>=51.0.0",
```
Comment (eli-schwartz) [inline pyproject.toml:5]:
```suggestion
    "setuptools>=61.0.0",
```

You *must* use a version of setuptools new enough to support using pyproject.toml metadata. This is the first version in which experimental support was added -- I have not checked if that is even sufficient, older versions may be buggy enough that you need to require an even higher minimum version, but this is the lowest version that it's even worth attempting to look at.

**Review by eli-schwartz — CHANGES_REQUESTED** [review]

File: pyproject.toml [hunk 2]
```diff
@@ -5,12 +5,145 @@ requires = [
...
     "wheel",
     "Cython>=0.29.32,<3",  # Note: sync with setup.py, environment.yml and asv.conf.json
-    "oldest-supported-numpy>=2022.8.16"
+    "oldest-supported-numpy>=2022.8.16",
+    "tomli; python_version < '3.11'" # Versioneer needs this
```
Comment (fangchenli) [inline pyproject.toml:9]: You can leave this out. #49924 add `versioneer[toml]` to pyproject.toml

File: pyproject.toml (hunk 1 above)
Comment (lithomas1) [reply pyproject.toml]: Thanks for the catch.

File: pyproject.toml (hunk 2 above)
Comment (lithomas1) [reply pyproject.toml:9]: I'll update after #49924 is merged.

**Review by fangchenli — APPROVED** [review]

**Review by WillAyd — APPROVED** [review]

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @lithomas1

**Pull request merged**